c7n/commands.py

```diff
@@ -526,6 +530,18 @@ def _print_cls_schema(cls):
     return
 
 
+def _expand_schema(schema, definitions):
+    for k, v in schema.items():
+        if k == '$ref':
+            schema.pop(k)
+            path = '.'.join(v.split('/')[2:])
+            return jmespath.search(path, definitions)
```

Comment:
make copy schema.items() else you'll get mutation during iteration issues
ditch the schema.pop()
`schema[k] = jmespath.search(..)`
afaics, no return else its early exit on key eval